MetaGenerator/UserInput/ValidatedUserInput.cs

```diff
@@ -47,14 +40,22 @@ public void WriteQuestion(IUserInputDescription variable, string defaultQuestion
             _consoleWriter.WriteLine(FormatWithVariableName(question, variable.VariableName));
         }
 
+        private Func<string, Option<string>> EchoAndValidate(IUserInputDescription variable, IEnumerable<IValidation> requesterValidations)
```

Comment:
Why don't you split this into an echo function and a validate function?

Reply:
Addressed in 531556de